src/main/java/com/example/nlearn/controllers/TestController.java

```diff
@@ -0,0 +1,119 @@
+package com.example.nlern.controllers;
+
+import com.example.nlern.models.FullTest;
+import com.example.nlern.models.Question;
+import com.example.nlern.models.Test;
+
+import com.example.nlern.repos.GroupRepository;
+import com.example.nlern.repos.QuestionRepository;
+import com.example.nlern.repos.TestRepository;
+import com.example.nlern.repos.UserRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.DeleteMapping;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/tests")
+public class TestController {
+
+    @Autowired
+    private QuestionRepository questionRepository;
```

Comment:
Same here, move this logic to services

Reply:
Done

Reply:
You missed this controller I suppose. Rewrite all logic from controllers to move it into respective services